src/transformers/models/llava_onevision/image_processing_llava_onevision_fast.py

```diff
@@ -0,0 +1,534 @@
+
+        image_patches = [resized_original_image] + patches
+
+        return image_patches
+
+    # Copied from transformers.models.llava_next.image_processing_llava_next_fast.LlavaNextImageProcessorFast._pad_for_batching
+    def _pad_for_batching(
+        self,
+        pixel_values: List["torch.Tensor"],
+    ) -> List["torch.Tensor"]:
+        """
+        Pads images on the `num_of_patches` dimension with zeros to form a batch of same number of patches.
+
+        Args:
+            pixel_values (`List[torch.Tensor]`):
+                An array of pixel values of each images of shape (`batch_size`, `num_patches`, `image_in_3D`)
+
+        Returns:
+            List[`torch.Tensor`]: The padded images.
+        """
+        max_patch = max(len(x) for x in pixel_values)
+        pixel_values = [
+            torch.nn.functional.pad(image, pad=[0, 0, 0, 0, 0, 0, 0, max_patch - image.shape[0]])
+            for image in pixel_values
+        ]
+
+        return pixel_values
+
+    def _build_transforms(
+        self,
+        do_resize: bool,
+        size: SizeDict,
+        interpolation: "F.InterpolationMode",
+        do_center_crop: bool,
+        crop_size: int,
+        do_rescale: bool,
+        rescale_factor: float,
+        do_normalize: bool,
+        image_mean: Union[float, List[float]],
+        image_std: Union[float, List[float]],
+    ) -> "Sequential":
+        """
+        Given the input settings build the image transforms using a `Sequential` module.
+        """
+        transforms = []
+
+        transforms.append(GroupByShape())
+        if do_resize:
+            transforms.append(Resize(size, interpolation=interpolation))
+            # Since the size was changed, we need to group the images by shape again
+            transforms.append(ReorderImages())
+            transforms.append(GroupByShape())
+        if do_center_crop:
+            transforms.append(CenterCrop(crop_size))
+            # Since the size was changed, we need to group the images by shape again
+            transforms.append(ReorderImages())
+            transforms.append(GroupByShape())
+        # We can combine rescale and normalize into a single operation for speed
+        if do_rescale and do_normalize:
+            # image_mean and image_std have already been adjusted for rescaling
+            transforms.append(Normalize(image_mean, image_std))
+        elif do_rescale:
+            transforms.append(Rescale(rescale_factor=rescale_factor))
+        elif do_normalize:
+            transforms.append(Normalize(image_mean, image_std))
+
+        if isinstance(transforms[-1], GroupByShape):
+            # No added transforms, so we can remove the last GroupByShape
+            transforms.pop()
+        else:
+            # We necessarily have grouped images, so we need to reorder them back to the original order
+            transforms.append(ReorderImages())
+
+        return Sequential(*transforms)
```

Comment:
Ok, this one looks a bit overcomplicated compared to the functional one, but I like that we have a separate method, which is easier to read. If we avoid the `GroupByShape` transform, it will look better. Instead, we can add `StackIfPossible`, which will stack images only if they all have the same size.

Reply:
It would be a shame to miss out on some nice speedups though. For example, with processors using patching, this group by shape method can lead to a 10 to 20x speedup. This would not work with something like StackIfPossible, as the border patches are usually of a different size than the middle patches.